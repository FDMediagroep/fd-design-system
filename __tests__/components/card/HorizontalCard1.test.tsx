import React from 'react';
import {
    HorizontalCard1,
    getCssClassNames,
} from '../../../src/components/card/HorizontalCard1';
import { render } from '@testing-library/react';
describe('HorizontalCard1', () => {
    test('should render correctly', () => {
        const { container, getByText } = render(
            <HorizontalCard1
                id="1"
                imageUrl="https://images.example.com/image.png"
                label="Ondernemen"
                title="This is the title"
                time="11:40"
                url="https://example.com/1/This is the title"
            />
        );

        expect(getByText('This is the title')).toBeTruthy();
        expect(getByText('Ondernemen')).toBeTruthy();
        expect(getByText('11:40')).toBeTruthy();
        expect(container.querySelector('[href]')?.getAttribute('href')).toBe(
            'https://example.com/1/This is the title'
        );
    });

    test('should render is-read correctly', () => {
        const { container } = render(
            <HorizontalCard1
                id="1"
                imageUrl="https://images.example.com/image.png"
                label="Ondernemen"
                title="This is the title"
                time="11:40"
                url="https://example.com/1/This is the title"
                isRead={true}
            />
        );
        expect(container).toMatchSnapshot();
    });

    test('should render variant-1 correctly', () => {
        const { container } = render(
            <HorizontalCard1
                id="1"
                imageUrl="https://images.example.com/image.png"
                label="Ondernemen"
                title="This is the title"
                time="11:40"
                url="https://example.com/1/This is the title"
                variant="variant-1"
            />
        );
        expect(container).toMatchSnapshot();
    });

    test('should have correct class name', () => {
        const { container } = render(
            <HorizontalCard1
                id="1"
                imageUrl="https://images.example.com/image.png"
                label="Ondernemen"
                title="This is the title"
                time="11:40"
                url="https://example.com/1/This is the title"
            />
        );

        expect(
            container?.firstElementChild?.classList.contains(
                getCssClassNames()[0]
            )
        ).toBeTruthy();
    });
});
