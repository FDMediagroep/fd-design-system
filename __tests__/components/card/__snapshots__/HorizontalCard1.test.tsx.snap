// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HorizontalCard1 should render is-read correctly 1`] = `
<div>
  <article
    class="undefined undefined xs__p+4"
    id="1"
  >
    <a
      aria-label="This is the title"
      href="https://example.com/1/This is the title"
    >
      <figure
        class="xs__m-0"
      >
        <picture>
          <img
            alt="This is the title"
            src="https://images.example.com/image.png"
          />
        </picture>
        <figcaption />
      </figure>
      <div
        class="undefined xs__pl+4 s__p+4"
      >
        <div
          class="undefined xs__m-0 xs__mb+2 body-text sans xs"
        >
          <time>
            11:40
          </time>
          <span>
            •
          </span>
          <span>
            Ondernemen
          </span>
        </div>
        <div>
          <h1
            class="heading serif s xs__m-0"
          >
            This is the title
          </h1>
        </div>
      </div>
    </a>
  </article>
</div>
`;

exports[`HorizontalCard1 should render variant-1 correctly 1`] = `
<div>
  <article
    class="undefined xs__p+4"
    id="1"
  >
    <a
      aria-label="This is the title"
      href="https://example.com/1/This is the title"
    >
      <figure
        class="xs__m-0"
      >
        <picture>
          <img
            alt="This is the title"
            src="https://images.example.com/image.png"
          />
        </picture>
        <figcaption />
      </figure>
      <div
        class="undefined xs__pl+4 s__p+4"
      >
        <div
          class="undefined xs__m-0 xs__mb+2 body-text sans xs"
        >
          <time>
            11:40
          </time>
          <span>
            •
          </span>
          <span>
            Ondernemen
          </span>
        </div>
        <div>
          <h1
            class="heading serif s xs__m-0"
          >
            This is the title
          </h1>
        </div>
      </div>
    </a>
  </article>
</div>
`;
